core/app/models/concerns/spree/display_link.rb

```diff
@@ -4,6 +4,7 @@ def link
       case linked_resource_type
       when 'Spree::Taxon'
         return if linked_resource.nil?
+        return if linked_resource.permalink.blank?
```

Comment:
```suggestion
        return if linked_resource&.permalink.blank?
```

I think we can simplify this using the safe operator